sencha-workspace/packages/slate-cbl/src/view/teacher/DashboardController.js

```diff
@@ -222,11 +242,145 @@ Ext.define('Slate.cbl.view.teacher.DashboardController', {
             });
         });
     },
+    
+    onDemonstrationDelete: function(demonstration, operation) {
+        var me = this,
+            dashboardView = me.view,
+            mainGridEl = dashboardView.el.down('.cbl-grid-main'),
+            studentId = demonstration.get('StudentID'),
+            competenciesStore = Ext.getStore('cbl-competencies-loaded'),
+            skillStudentsTpl = dashboardView.getTpl('skillStudentsTpl'),
+
+            deletedSkills = demonstration.get('Skills'),
+            deletedSkillsLength = deletedSkills.length, deletedSkillIndex = 0, demonstratedSkill,
+            loadedCompetency, loadedSkill, demonstrationsByStudent, skillDemonstrationsCell, deletedSkillIds, studentDemonstrations, studentDemonstrationIndex = 0,
+
+            competencyCompletions = operation.getProxy().getReader().rawData.data[0].competencyCompletions,
+            competencyCompletionsLength = competencyCompletions.length, competencyCompletionIndex = 0, competencyCompletion,
+            competencyProgressCell, competencyPercent,
+            _compentencyDemonstrationResetFn,
+            _updateCompetencySkillsFn, 
+
+            updatesCompetencies = [],
+            updatesLength, updateIndex, update, skillsRows, skillHeadersRow, skillStudentsRow, competencySkills, skill,
+            
+            updatedDemonstrationLength, updatedDemonstrationIndex;
+
+
+        // compile operations and DOM references for needed updates -- don't write to dom!
```

Comment:
What sort of context are you looking for? This is just the general best practice that heavy DOM manipulation should be batched into read and write phases. This is the read phase. Bouncing between reads and writes triggers reflows.